### DISCUSSION TITLE
Upgrade to 2.16.0 and above give failed tests

### DISCUSSION THREAD

**Discussion by kixdcrwtkqgiyrxs** [Royal-Navy/design-system#1625] [Q&A] [answered]
TypeError: Cannot read property 'getAllByTestId' of undefined
```
expect(
    wrapper.getAllByTestId('data-testid')
)
```

where 

```
wrapper = render(
    <Router>
        <SomeFormComponent>
    </Router>
)
```

render imported from `@testing-library/react` and Router is MemoryRouter imported from `react-router-dom`

**Comment by kixdcrwtkqgiyrxs** [comment]
Looks like it's do to with number inputs specifically :)

**Comment by thyhjwb6** [comment]
I am attempting to reproduce this:

```javascript
  describe('Repro', () => {
    it('should render', () => {
      wrapper = render(
        <Router>
          <NumberInput
            data-testid="blah"
            name="number-input"
            onChange={onChangeSpy}
          />
        </Router>
      )

      expect(wrapper.getAllByTestId('blah')).toHaveLength(1)
    })
  })
```

The test passes. Are you able to provide a failing test?

**Comment by thyhjwb6** [comment, marked as answer]
This is because `NumberInput` is dependent on the canvas API. We have worked around this by adding `import 'jest-canvas-mock'` to `setupTests.ts`.
